test/gpu/octree/test_radius_search.cpp

```diff
@@ -135,25 +145,43 @@ TEST(PCL_OctreeGPU, batchRadiusSearch)
     result_device1.data.download(downloaded_buffer1);
     result_device2.data.download(downloaded_buffer2);
     result_device3.data.download(downloaded_buffer3);
+
+    std::vector<float> sqr_distances1, sqr_distances2, sqr_distances3;
+    result_sqr_distances1.download(sqr_distances1);
+    result_sqr_distances2.download(sqr_distances2);
+    result_sqr_distances3.download(sqr_distances3);
         
     //data.bruteForceSearch();
 
     //verify results    
     for(std::size_t i = 0; i < data.tests_num; ++i)
     {        
         std::vector<int>& results_host = host_search1[i];        
-        
+        std::vector<float>& sqr_dist_host = host_sqr_distances1[i];
+
         int beg = i * max_answers;
         int end = beg + sizes1[i];
 
+        const auto beg_dist2 = sqr_distances1.cbegin() + beg;
+        const auto end_dist2 = sqr_distances1.cbegin() + end;
+
         results_batch.assign(downloaded_buffer1.begin() + beg, downloaded_buffer1.begin() + end);
+        std::vector<float> sqr_distances_batch (beg_dist2, end_dist2);
 
         std::sort(results_batch.begin(), results_batch.end());
         std::sort(results_host.begin(), results_host.end());
 
         if ((int)results_batch.size() == max_answers && results_batch.size() < results_host.size() && max_answers)
+        {
             results_host.resize(max_answers);
-        
+            sqr_dist_host.resize(max_answers);
+        }
+        const float sqr_radius = data.shared_radius * data.shared_radius;
+        for (std::size_t j = 0; j < sqr_dist_host.size(); j++)
+        {
+            ASSERT_LT (sqr_distances_batch[j], sqr_radius);
+            ASSERT_NEAR ( sqr_distances_batch[j], sqr_dist_host[j], 0.001);
+        }
         ASSERT_EQ ( ( results_batch == results_host ), true );       
```

Comment:
You need to refactor this. There's a lot of code repetition. 
https://github.com/PointCloudLibrary/pcl/blob/c762664e9a4ddb08075985b1edbf1565a00777e0/test/gpu/octree/test_radius_search.cpp#L149-L263